jsf/jsf-widgets/src/java/org/sakaiproject/jsf/renderer/ViewRenderer.java

```diff
@@ -77,6 +84,14 @@ public void encodeBegin(FacesContext context, UIComponent component) throws IOEx
 
             String lang = locale.getLanguage();
 
+            String thisUser = SessionManager.getCurrentSessionUserId();
+            PreferencesService preferencesService = ComponentManager.get(PreferencesService.class);
+            Preferences prefs = preferencesService.getPreferences(thisUser);
+            String userTheme = "sakaiUserTheme-notSet";
+            if ( prefs != null ) {
+                userTheme = StringUtils.defaultIfEmpty(prefs.getProperties(org.sakaiproject.user.api.PreferencesService.USER_SELECTED_UI_THEME_PREFS).getProperty("theme"), "sakaiUserTheme-notSet");
+            }
+
```

Comment:
I'd suggest we wrap all of this with a condition `if(sakaiThemesEnabled)` that checks for the "portal.themes" property: `sakaiThemesEnabled = ServerConfigurationService.getBoolean("portal.themes", true);` since there is no need to check the preference if themes are disabled in the instance.

Reply:
Yup - I will do this.

Reply:
Check out the latest.